src/FM.LiveSwitch.Mux.Standard/FfmpegUtility.cs

```diff
@@ -0,0 +1,121 @@
+﻿using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace FM.LiveSwitch.Mux
+{
+    public class FfmpegUtility
```

Comment:
I think we should make a few changes here.

First, let's rename the class to `Utility`. The methods inside are already prefixed with `FFmpeg` or `FFprobe`.

Second, let's move the `GetDuration` method out of here and into `JsonPreprocessor` which is the only place it is used.

Third, the `fileName` parameter in `GetDuration` should be renamed to `filePath`.

Fourth, we need to get rid of the file-extension-based detection of audio vs. video and turn it into a method parameter. The file extension could be anything. We don't really care what it is. Audio in a .mkv file is entirely possible, and since the caller already knows if the file is audio or video, we can tighten this up:

```
public Task<TimeSpan?> GetAudioDuration(string filePath)
{
    return GetDuration(filePath, true);
}

public Task<TimeSpan?> GetVideoDuration(string filePath)
{
    return GetDuration(filePath, false);
}

private async Task<TimeSpan?> GetDuration(string filePath, bool audio)
{
    var option = audio ? "a" : "v";
    var lines = ...;
    ...
}
```

This way any upstream changes in the Media Server will never have an impact here.